strings/longest_substr_without_repeating_chars.cpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file
+ * @brief Greedy algorithm to find the [length of the longest substring without
+ * repeating
+ * characters](https://www.geeksforgeeks.org/length-of-the-longest-substring-without-repeating-characters/)
+ *
+ *
+ * @details
+ * We are given a string. We need to compute the length of the longest
+ * substring, such that no character in the substring occurs more than once.
+ *
+ * For a string of length n, the time complexity of this algorithm is O(n)
+ * (amortized)
+ *
+ * @author [Nishant Mahesh](https://github.com/nishant-mahesh)
+ */
+
+#include <algorithm>  // To use the inbuilt algorithmic functions, eg: 'max'
+#include <iostream>   // For IO Operations
+#include <map>     // Used to efficiently store and retrieve (key, value) pairs
+#include <string>  // For string operations
+
+using std::cin;
+using std::cout;
+using std::map;
+using std::max;
+using std::pair;
+using std::string;
+
+/**
+ * @param string named "input"
```

Comment:
Yes, we know it's a string named `input`, but what is the parameter used for? This needs to be documented. :)
Let us know if you need any help with anything!